### ISSUE TITLE
elasticsearch-5.6.6-SNAPSHOT.deb cannot access ‘/etc/elasticsearch/scripts’: No such file or directory

### ISSUE THREAD

**Issue by LeeDr** [elastic/elasticsearch#28141]
**Elasticsearch version** (`bin/elasticsearch --version`): 5.6.6-SNAPSHOT

**Plugins installed**: []

**JVM version** (`java -version`): openjdk version "1.8.0_141"
OpenJDK Runtime Environment (build 1.8.0_141-8u141-b15-3~14.04-b15)
OpenJDK 64-Bit Server VM (build 25.141-b15, mixed mode)

**OS version** (`uname -a` if on a Unix-like system):  (VM box "elastic/ubuntu-14.04-x86_64")
Linux packer-virtualbox-iso-1514734280 4.4.0-104-generic #127~14.04.1-Ubuntu SMP Mon Dec 11 12:44:15 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**:

Elasticsearch fails to install with error;
 `chmod: cannot access ‘/etc/elasticsearch/scripts’: No such file or directory`

**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem, including
(e.g.) index creation, mappings, settings, query etc.  The easier you make for
us to reproduce it, the more likely that somebody will take the time to look at it.

 1. Download http://snapshots.elastic.co/downloads/elasticsearch/elasticsearch-5.6.6-SNAPSHOT.deb
 2. as root user on Ubuntu 14.04 install it with `dpkg -i elasticsearch-5.6.6-SNAPSHOT.deb`
 3. install fails with error

**Provide logs (if relevant)**:

```
==> ubuntu_deb_no_xpack: ++ dpkg -i ./elasticsearch-5.6.6-SNAPSHOT.deb
==> ubuntu_deb_no_xpack: Selecting previously unselected package elasticsearch.
==> ubuntu_deb_no_xpack: (Reading database ... 144965 files and directories currently installed.)
==> ubuntu_deb_no_xpack: Preparing to unpack .../elasticsearch-5.6.6-SNAPSHOT.deb ...
==> ubuntu_deb_no_xpack: Creating elasticsearch group...
==> ubuntu_deb_no_xpack:  OK
==> ubuntu_deb_no_xpack: Creating elasticsearch user...
==> ubuntu_deb_no_xpack:  OK
==> ubuntu_deb_no_xpack: Unpacking elasticsearch (5.6.6-SNAPSHOT) ...
==> ubuntu_deb_no_xpack: Setting up elasticsearch (5.6.6-SNAPSHOT) ...
==> ubuntu_deb_no_xpack: chmod: cannot access ‘/etc/elasticsearch/scripts’: No such file or directory
==> ubuntu_deb_no_xpack: dpkg: error processing package elasticsearch (--install):
==> ubuntu_deb_no_xpack:  subprocess installed post-installation script returned error exit status 1
==> ubuntu_deb_no_xpack: Processing triggers for ureadahead (0.100.0-16) ...
==> ubuntu_deb_no_xpack: Errors were encountered while processing:
==> ubuntu_deb_no_xpack:  elasticsearch
```

**Comment by LeeDr** [comment]
The .rpm package does install on CentOS 7, but then fails to start up;

```
[root@localhost vagrant]# cat /var/log/elasticsearch/elasticsearch.log
[2018-01-09T00:08:35,561][ERROR][o.e.b.Bootstrap          ] Exception
java.lang.IllegalStateException: Unable to access 'path.scripts' (/etc/elasticsearch/scripts)
        at org.elasticsearch.bootstrap.Security.addPath(Security.java:450) ~[elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.bootstrap.Security.addFilePermissions(Security.java:282) ~[elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.bootstrap.Security.createPermissions(Security.java:246) ~[elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.bootstrap.Security.configure(Security.java:119) ~[elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:228) ~[elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:342) [elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:132) [elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:123) [elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:70) [elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:134) [elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.cli.Command.main(Command.java:90) [elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:91) [elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:84) [elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
Caused by: java.nio.file.AccessDeniedException: /etc/elasticsearch/scripts
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84) ~[?:1.8.0_151]
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102) ~[?:1.8.0_151]
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107) ~[?:1.8.0_151]
        at sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:384) ~[?:1.8.0_151]
        at java.nio.file.Files.createDirectory(Files.java:674) ~[?:1.8.0_151]
        at java.nio.file.Files.createAndCheckIsDirectory(Files.java:781) ~[?:1.8.0_151]
        at java.nio.file.Files.createDirectories(Files.java:767) ~[?:1.8.0_151]
        at org.elasticsearch.bootstrap.Security.ensureDirectoryExists(Security.java:492) ~[elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        at org.elasticsearch.bootstrap.Security.addPath(Security.java:448) ~[elasticsearch-5.6.6-SNAPSHOT.jar:5.6.6-SNAPSHOT]
        ... 12 more
```

**Comment by jasontedor** [comment]
Closed by 80e78e9ecc0d409cbf32343b3e738109e90019fd
